contracts/core/src/token/OmniGasPump.sol

```diff
@@ -0,0 +1,204 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity 0.8.24;
+
+import { OwnableUpgradeable } from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { PausableUpgradeable } from "@openzeppelin/contracts-upgradeable/utils/PausableUpgradeable.sol";
+import { XAppUpgradeable } from "src/pkg/XAppUpgradeable.sol";
+import { FeeOracleV1 } from "src/xchain/FeeOracleV1.sol";
+import { ConfLevel } from "src/libraries/ConfLevel.sol";
+import { OmniXFund } from "./OmniXFund.sol";
+
+/**
+ * @title OmniGasPump
+ * @notice A unidirectional cross-chain gas exchange, allowing users to swap native ETH for native OMNI.[]
+ */
+contract OmniGasPump is XAppUpgradeable, OwnableUpgradeable, PausableUpgradeable {
+    event MaxSwapSet(uint256 max);
+    event FundSet(address fund);
+    event PctCutSet(uint256 pct);
+    event Refilled(address indexed recipient, uint256 paid, uint256 fee, uint256 cut, uint256 amtOMNI);
+
+    /// @notice Gas limit passed to OmniXFund.settleUp xcall
+    uint64 internal constant SETTLE_GAS = 100_000;
+
+    /// @notice Denominator for pct cut
+    uint256 internal constant PCT_DENOM = 1000;
```

Comment:
1_000

Reply:
Form fmt does not request underscores for this so gonna leave it